transport-native-epoll/src/main/java/io/netty/channel/epoll/NativeDatagramPacketArray.java

```diff
@@ -145,10 +150,11 @@ private void init(long memoryAddress, int count, InetSocketAddress recipient) {
 
         DatagramPacket newDatagramPacket(ByteBuf buffer, InetSocketAddress localAddress) throws UnknownHostException {
             final InetAddress address;
-            if (scopeId != 0) {
-                address = Inet6Address.getByAddress(null, addr, scopeId);
+            if (addrLen == ipv4Bytes.length) {
+                System.arraycopy(addr, 0, ipv4Bytes, 0, addrLen);
```

Comment:
Don't we need to sync on ipv4Bytes over these two operations so it's thread safe?

Reply:
Nope all the access here is single threaded